R/calc_plan_stats.R

```diff
@@ -0,0 +1,88 @@
+calc_plan_stats <- function(redist_plans, redist_map, ...) {
+
+    redist_plans <- redist_plans %>%
+        mutate(total_vap = tally_var(redist_map, vap),
+               plan_dev =  plan_parity(redist_map),
+               comp_edge = distr_compactness(redist_map),
+               comp_polsby = distr_compactness(redist_map,
+                                               measure = "PolsbyPopper"),
+               ndv = tally_var(redist_map, ndv),
+               nrv = tally_var(redist_map, nrv),
+               ndshare = ndv / (ndv + nrv),
+               ...)
+
+    tally_cols <- names(redist_map)[c(tidyselect::eval_select(starts_with("pop_"), redist_map),
+                                      tidyselect::eval_select(starts_with("vap_"), redist_map),
+                                      tidyselect::eval_select(matches("_(dem|rep)_"), redist_map),
+                                      tidyselect::eval_select(matches("^a[dr]v_"), redist_map))]
+    for (col in tally_cols) {
+        redist_plans <- mutate(redist_plans, {{ col }} := tally_var(redist_map, redist_map[[col]]), .before = ndv)
+    }
+
+    elecs <- select(as_tibble(redist_map), contains("_dem_")) %>%
+        names() %>%
+        str_sub(1, 6) %>%
+        unique()
+
+    elect_tb <- purrr::map_dfr(elecs, function(el) {
```

Comment:
let's replace this `purrr::map_dfr` with `do.call(dplyr::bind_rows, lapply(elecs, function(el) { ... }))` which does the same thing but lets us not have to import the `purrr` package